### PR TITLE
add installation instructions to README.md

### PR DESCRIPTION
**Pull request by rogpeppe**
To QA, try to follow the instructions!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,12 @@
 # macarooncompat - compatibility checking for macaroons.
+
+To run, first install the following:
+
+-  [libsodium](http://doc.libsodium.org/)
```
Comment (frankban) [inline README.md:5]: Unnecessary space here.

**Comment by frankban** [comment]
This is a good start :+1: 
I'd be inclined to have explicit instructions here, something to guide future automation.
For instance, in pymacaroons the libsodium requirement is automated as:
- wget https://github.com/jedisct1/libsodium/releases/download/0.7.0/libsodium-0.7.0.tar.gz
- tar xzvf libsodium-0.7.0.tar.gz
- cd libsodium-0.7.0
- ./configure && make && make check && sudo make install
- sudo ldconfig
- cd ..
  I did not test the above, just used as an example.

**Comment by cmars** [comment]
I'd like something similar for node.js & jsmacaroon.

Longer term I think it'd be nice if we packaged these deps. I did a [deb](https://github.com/cmars/libsodium-debian) for libsodium-0.5 a while back, published [here](https://launchpad.net/~cmars/+archive/ubuntu/libsodium). Would love to get something similar going for libmacaroon and propose into vivid if there's still time.

**Comment by rogpeppe** [comment]
Landing this as a start, at least.

**Pull request merged**